x/capability/keeper/keeper.go

```diff
@@ -184,13 +207,17 @@ func (sk ScopedKeeper) ClaimCapability(ctx sdk.Context, cap *types.Capability, n
 		return err
 	}
 
+	memStore := ctx.KVStore(sk.memKey)
+
 	// Set the forward mapping between the module and capability tuple and the
-	// capability name in the in-memory store.
-	sk.capStore.SetCapabilityName(sk.module, name, cap)
+	// capability name in the memKVStore
+	memStore.Set(types.FwdCapabilityKey(sk.module, cap), []byte(name))
 
 	// Set the reverse mapping between the module and capability name and the
-	// capability in the in-memory store.
-	sk.capStore.SetCapability(sk.module, name, cap)
+	// index in the in-memory store. Since marshalling and unmarshalling into a store
+	// will change memory address of capability, we simply store index as value here
+	// and retrieve the in-memory pointer to the capability from our map
+	memStore.Set(types.RevCapabilityKey(sk.module, name), sdk.Uint64ToBigEndian(cap.GetIndex()))
 
```

Comment:
Do we not need to set the capability in the map?

Reply:
No, the capability already exists in the map when we call `ClaimCapability` since someone has already created that mapping `index => *cap` when they called `NewCapability` beforehand. We just need to set mappings in the memKVStore to record our own local name of that capability